torch/distributed/rpc/rref_proxy.py

```diff
@@ -15,24 +15,82 @@ def _local_invoke_async_execution(rref, func_name, args, kwargs):
 def _invoke_rpc(rref, rpc_api, func_name, timeout, *args, **kwargs):
     # Since rref._get_type can potentially issue an RPC, it should respect the
     # passed in timeout here.
-    rref_type = rref._get_type(timeout=timeout)
-
-    _invoke_func = _local_invoke
-    # Bypass ScriptModules when checking for async function attribute.
-    bypass_type = issubclass(rref_type, torch.jit.ScriptModule) or issubclass(
-        rref_type, torch._C.ScriptModule
-    )
-    if not bypass_type:
-        func = getattr(rref_type, func_name)
-        if hasattr(func, "_wrapped_async_rpc_function"):
-            _invoke_func = _local_invoke_async_execution
-
-    return rpc_api(
-        rref.owner(),
-        _invoke_func,
-        args=(rref, func_name, args, kwargs),
-        timeout=timeout
-    )
+    rref_type = rref._get_type(timeout=timeout, blocking=False)
+    # Controls whether we are running in a callback or inline in invoke_on_owner
+    # to appropriately handle errors.
+    block_on_type = rpc_api in [
```

Comment:
Still need this for branching in the callback, since it controls whether to throw the error or wrap it in a future. Although, not sure if there's a better way